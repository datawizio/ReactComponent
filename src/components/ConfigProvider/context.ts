import { createContext } from "react";

export interface ConfigContextValue {
  translate: (transKey: string) => string;
}

const ConfigContext = createContext<ConfigContextValue>({
  translate: transKey => transKey
});

export default ConfigContext;
