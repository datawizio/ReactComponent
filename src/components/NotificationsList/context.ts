import { createContext, Dispatch } from "react";
import { Action, IListItem, ListState } from "./types";

export const NotificationsListContext = createContext<{
  state: ListState<IListItem> | null;
  dispatch: Dispatch<Action<IListItem>> | null;
}>({ state: null, dispatch: null });

NotificationsListContext.displayName = "NotificationsListContext";
