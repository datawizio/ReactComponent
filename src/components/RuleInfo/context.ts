import { createContext } from "react";
import { IRuleInfoContext } from "./types";

export const RuleInfoContext = createContext<IRuleInfoContext>(null);

RuleInfoContext.displayName = "RuleInfoContext";
