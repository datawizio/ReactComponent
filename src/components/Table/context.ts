import React, { Dispatch } from "react";
import { TableProps, TableState, Action } from "./types";

export const TableContext = React.createContext<{
  tableState: TableState;
  tableProps: TableProps;
  baseTableState: TableState;
  dispatch: Dispatch<Action>;
}>(null);

TableContext.displayName = "TableContext";
