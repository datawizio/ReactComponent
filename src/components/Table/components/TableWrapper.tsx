import * as React from "react";
import { useRef, useState, useEffect, useMemo, useContext } from "react";
import resizeDetector from "../../../utils/resizeDetector";
import { getAbsoluteHeight } from "../../../utils/sizeUtils";
import { TableContext } from "../context";

const TableWrapper: React.FC<React.HTMLAttributes<any>> = props => {
  const resizeRef = useRef<() => void>();
  const wrapperRef = useRef<HTMLDivElement>();

  const { style, ...tableProps } = props;
  const { height: _height, width: _width, ...tableStyles } = style;

  const {
    tableState: { dataSource },
    tableProps: { responsiveTable, autoHeight }
  } = useContext(TableContext);

  const [width, setWidth] = useState<any>(_width);
  const [height, setHeight] = useState<any>(_height);

  useEffect(() => setWidth(_width), [_width]);
  useEffect(() => setHeight(_height), [_height]);

  useEffect(() => {
    if (wrapperRef.current && responsiveTable && dataSource.length) {
      const container = wrapperRef.current.closest(".dw-table-container");

      if (container && container.parentElement) {
        const parent = container.parentElement;
        const pagination = container.getElementsByClassName(
          "ant-pagination"
        )[0];

        const toolbar = container.getElementsByClassName(
          "dw-table__toolbar"
        )[0];

        resizeRef.current && resizeRef.current();

        resizeRef.current = resizeDetector(parent, (elHeight, elWidth) => {
          const offset = [toolbar, pagination].reduce((acc, el) => {
            return el ? acc + getAbsoluteHeight(el as any) : acc;
          }, 0);

          setWidth(elWidth);
          setHeight(elHeight - offset);
        });

        return resizeRef.current;
      }
    }
  }, [responsiveTable, wrapperRef, dataSource]);

  const styles = useMemo(() => {
    const _styles: React.CSSProperties = {
      width
    };

    if (autoHeight) _styles.maxHeight = height;
    else _styles.height = height;

    return _styles;
  }, [height, width, autoHeight]);

  return (
    <div style={styles} ref={wrapperRef} className="dw-table__wrapper">
      <table {...tableProps} style={tableStyles} />
    </div>
  );
};

export default TableWrapper;
