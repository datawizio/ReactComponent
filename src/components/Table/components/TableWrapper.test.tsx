import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";
import TableWrapper from "./TableWrapper";
import { TableContext } from "../context";

const context = {
  tableState: { dTypesConfig: [] },
  tableProps: {
    autoHeight: false,
    responsiveTable: false
  }
};

const componentWithContext = props => {
  return (
    //@ts-ignore
    <TableContext.Provider value={context}>
      <TableWrapper {...props} />
    </TableContext.Provider>
  );
};

const setUp = (props?) => mount(componentWithContext(props));

describe("TableWrapper component", () => {
  let component;

  beforeEach(() => {
    component = setUp({
      style: {
        height: 1,
        width: 1
      }
    });
  });

  it("rendered TableWrapper corectly", () => {
    expect(component).toMatchSnapshot();
  });

  it("TableWrapperBar has 1 cell class ", () => {
    const tableWrapperClass = component.find(".dw-table__wrapper");
    expect(tableWrapperClass.length).toBe(1);
  });

  it("TableWrapperlBar has table element", () => {
    const tableElement = component.find("table");
    expect(tableElement.length).toBe(1);
  });
});
