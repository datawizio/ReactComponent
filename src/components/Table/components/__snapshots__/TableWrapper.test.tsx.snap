// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableWrapper component rendered TableWrapper corectly 1`] = `
<TableWrapper
  style={
    Object {
      "height": 1,
      "width": 1,
    }
  }
>
  <div
    className="dw-table__wrapper"
    style={
      Object {
        "height": 1,
        "width": 1,
      }
    }
  >
    <table
      style={Object {}}
    />
  </div>
</TableWrapper>
`;
